zonemaster/zonemaster-engine
# Fix bug in Zone01

## Purpose

This PR proposes an update to Test Case Zone01 due to an oversight in #1035 which led to Engine crashing when several nameservers are found to serve different SOA serials.

## Context

Fixes #1175 

## Changes

- Fix missing library import
- Fix code for message tag Z01_MNAME_NOT_MASTER
- Add unit tests
- Add function `serial_gt` in Zonemaster::Engine::Util, and its documentation
- Refactoring

## How to test this PR

Unit tests should pass.

But if you want to test manually with zonemaster-cli, create a file (e.g. `local.hints`) with the following content:
```
$ more local.hints
.                              3600000      NS    IBDNS01.LABS.PRIVE.NIC.FR.
IBDNS01.LABS.PRIVE.NIC.FR.     3600000      A     10.1.72.23
.                              3600000      NS    IBDNS01-24.LABS.PRIVE.NIC.FR.
IBDNS01-24.LABS.PRIVE.NIC.FR.  3600000      A     10.1.72.24
```
Then you can do:
```
$ zonemaster-cli --test zone/zone01 --hints local.hints mname-not-master.zone01.xa --level=INFO
Seconds Level     Message
======= ========= =======
   0.00 INFO      Using version v4.6.0 of the Zonemaster engine.
   0.07 WARNING   SOA MNAME name server(s) "ns1.mname-not-master.zone01.xa/10.1.72.23" do not have the highest SOA SERIAL (expected "2022122300" but got "2022122300;2022122301")
```

Also, taken from https://github.com/zonemaster/zonemaster-engine/issues/1175#issuecomment-1380227232:
(note that I disable IPv6 due to my machine limitations at the moment)
```
$ zonemaster-cli --test zone/zone01 hwcdn.net --level=INFO --no-ipv6
Seconds Level     Message
======= ========= =======
   0.00 INFO      Using version v4.6.0 of the Zonemaster engine.
   0.56 WARNING   SOA MNAME name server(s) "ns1.hwcdn.net/69.16.174.10" do not have the highest SOA SERIAL (expected "1673538139" but got "1673538139;1673538142")
```
